File: 프로그래머스/lv1/42748. K번째수/K번째수.js
function solution(array, commands) {
    let answer = [];
    
    commands.forEach((e) => {
        answer.push(
            array.filter((el, idx) => { if (idx >= e[0] - 1 && idx <= e[1] - 1) return el})
            .sort((a, b) => a - b)[e[2] - 1]
        )
    })
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer> answer = new ArrayList<>();
                
        for (int[] c : commands) {
            int start = c[0];
            int end = c[1];
            int idx = c[2] - 1;
            
            int[] sub = Arrays.copyOfRange(array, start - 1, end);
            
            Arrays.sort(sub);
            answer.add(sub[idx]);
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}